metrics-exporter-prometheus/src/builder.rs

```diff
@@ -531,6 +547,28 @@ impl Default for PrometheusBuilder {
     }
 }
 
+fn basic_auth<U, P>(username: U, password: Option<P>) -> HeaderValue
+where
+    U: std::fmt::Display,
+    P: std::fmt::Display,
+{
+    use base64::prelude::BASE64_STANDARD;
+    use base64::write::EncoderWriter;
+    use std::io::Write;
+
+    let mut buf = b"Basic ".to_vec();
+    {
+        let mut encoder = EncoderWriter::new(&mut buf, &BASE64_STANDARD);
+        let _ = write!(encoder, "{username}:");
+        if let Some(password) = password {
+            let _ = write!(encoder, "{password}");
+        }
+    }
+    let mut header = HeaderValue::from_bytes(&buf).expect("base64 is always valid HeaderValue");
+    header.set_sensitive(true);
+    header
+}
+
```

Comment:
This should have a simple unit test below that ensures the output matches well-known examples for both username only, and username/password.

Reply:
And also that if the return value (`HeaderValue`) is rendered via its debug representation (`{:?}`), the value is `Sensitive`.

(I realize it's very obvious from the code that this should happen, but since we're not using `typed-headers`, I want to make sure we're validating basic properties via tests)